src/info/index_error.h

```diff
@@ -30,16 +35,27 @@ extern char* const IndexError_ObjectName;
 IndexError IndexError_Init();
 
 // Adds an error message to the IndexError. The error_count is incremented and the last_error is set to the error_message.
-void IndexError_AddError(IndexError *error, const char *error_message, RedisModuleString *key);
+void IndexError_AddError(IndexError *error, ConstErrorMessage withoutUserData, ConstErrorMessage withUserData, RedisModuleString *key);
+
+// Adds a query error to the index error using IndexError_AddError
+static inline void IndexError_AddQueryError(IndexError *error, const QueryError* queryError, RedisModuleString *key) {
```

Comment:
Can you add a comment about the difference between this and `IndexError_AddError`?